docs/core/diagnostics/observability-with-otel.md

```diff
@@ -0,0 +1,429 @@
+---
+title: .NET Observability with OpenTelemetry
+description: An introduction to observing .NET apps with OpenTelemetry
+ms.date: 6/14/2023
+ms.topic: conceptual
+---
+
+# .NET observability with OpenTelemetry
+
+When you run a distributed application, you want to know how well the app is performing and to detect potential problems before they become larger. One way that you can do this is by monitoring specific instrumentation emitted by the application and looking for changes in metrics and logs.
+
+## What is observability
+
+Observability in the context of a distributed system is the ability to monitor and analyze telemetry about the state of each component, to be able to observe changes in performance, and to diagnose why those changes occur. Unlike debugging, which is invasive and can affect the operation of the application, observability is intended to be transparent to the primary operation and have a small enough performance impact that it can be used continuously.
+
+Observability is commonly done using a combination of:
+
+- Logs - record individual operations, such as an incoming request, a failure in a specific component, or an order being placed.
+- Metrics - measuring counters and gauges such as number of completed requests, active requests, widgets that have been sold; or a histogram of the request latency.
+- Distributed tracing - track requests and activities across components in a distributed system so that you can see where time is spent and track down specific failures.
+
+Together, logs, metrics, and distributed tracing are known as the *3 pillars of observability*.
+
+Each pillar might include telemetry data from:
+
+- The .NET runtime, such as the garbage collector or JIT.
+- Libraries, such as from Kestrel (the ASP.NET web server) and `HttpClient`.
+- Application-specific telemetry that's fired by your code.
+
+## Observability approaches in .NET
+
+There are a few different ways to achieve observability in .NET applications:
+
+- Explicitly in code, by referencing and using an SDK such as OpenTelemetry. If you have access to the source code and can rebuild the app, then this is the most powerful and configurable mechanism.
+- Out-of-process using [EventPipe](./eventpipe.md). Tools such as [dotnet-monitor](./dotnet-monitor.md) can listen to logs and metrics and then process them without affecting any code.
+- [Using a startup hook](https://github.com/dotnet/runtime/blob/main/docs/design/features/host-startup-hook.md), assemblies can be injected into the process that can then collect instrumentation. This mechanism enables creating instrumentation for libraries that weren't designed for it but can be fragile to version conflicts between application and instrumentation dependencies. An example of this approach is [OpenTelemetry .NET Automatic Instrumentation](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/README.md).
```

Comment:
I want to help direct people to .NET monitor instead - and they need to start to understand why. Otherwise they will ask why do all this stuff if it can happen automatically for me.

Reply:
We can include a link to dotnet-monitor in the bullet for code-less instrumentation without needing to dive into implementation details.